http.js

```
@@ -393,11 +392,11 @@ exports.read = function (request, qualifier) {
             if (Buffer.isBuffer(data)) {
                 contentType = response.headers['content-type'] || '',
                 match = contentType.match(/charset=([a-z0-9-]+)/i),
-                charset = match ? match[1] : 'utf8';
+                charset = match && iconv.encodingExists(match[1]) ? match[1] : 'utf8';
```

And perhaps we should throw an error if the encoding does not exist.